docker-wrappers/SPRAS/README.md

```diff
@@ -48,16 +48,66 @@ DOCKER_DEFAULT_PLATFORM=linux/amd64 docker build -t reedcompbio/spras:v0.1.0 -f
 The folder `docker-wrappers/SPRAS` also contains several files that can be used to test this container on HTCondor. To test the `spras` container
 in this environment, first login to an HTCondor Access Point (AP). Then, from the AP clone this repo:
 
-```
+```bash
 git clone https://github.com/Reed-CompBio/spras.git
 ```
 
-When you're ready to run SPRAS as an HTCondor workflow, navigate to the `spras/docker-wrappers/SPRAS` directory and create the `logs/` directory. Then run
-`condor_submit spras.sub`, which will submit SPRAS to HTCondor as a single job with as many cores as indicated by the `NUM_PROCS` line in `spras.sub`, using
-the value of `EXAMPLE_CONFIG` as the SPRAS configuration file. Note that you can alter the configuration file to test various workflows, but you should leave
-`unpack_singularity = true`, or it is likely the job will be unsuccessful. By default, the `example_config.yaml` runs everything except for `cytoscape`, which
-appears to fail periodically in HTCondor.
+There are currently two options for running SPRAS with HTCondor. The first is to submit all SPRAS jobs to a single remote Execution Point (EP). The second
+is to use the the snakemake HTCondor executor to parallelize the workflow by submitting each job to its own EP.
+
+### Submitting All Jobs to a Single EP
+
+Navigate to the `spras/docker-wrappers/SPRAS` directory and create the `logs/` directory. Then run `condor_submit spras.sub`, which will submit SPRAS
+to HTCondor as a single job with as many cores as indicated by the `NUM_PROCS` line in `spras.sub`, using the value of `EXAMPLE_CONFIG` as the SPRAS
+configuration file. Note that you can alter the configuration file to test various workflows, but you should leave `unpack_singularity = true`, or it
+is likely the job will be unsuccessful. By default, the `example_config.yaml` runs everything except for `cytoscape`, which appears to fail periodically
+in HTCondor.
+
+**Note**: The `spras.sub` submit file is an example of how this workflow could be submitted from a CHTC Access Point (AP) to the OSPool. To run in the local
+CHTC pool, omit the `+WantGlideIn` and `requirements` lines
+
+### Submitting Parallel Jobs
+
+Parallelizing SPRAS workflows with HTCondor currently requires an experimental executor for HTCondor that has been forked from the upstream [HTCondor Snakemake executor](https://github.com/jhiemstrawisc/snakemake-executor-plugin-htcondor/tree/spras-feature-dev).
+To get this executor, clone the forked repository using the following:
+```bash
+git clone -b spras-feature-dev https://github.com/jhiemstrawisc/snakemake-executor-plugin-htcondor.git
+```
+
+Then, from your activated `spras` conda environment (important), run:
+```bash
+pip install snakemake-executor-plugin-htcondor/
```

Comment:
Should we add this directly to the conda environment file? It is possible to pip install directly from GitHub, and I believe we could also specify the branch. That would simplify the setup steps.

Reply:
I tried doing that in another commit to this PR, but it doesn't seem like the `htcondor` python bindings needed by the executor play well with arm64/mac or Windows. If we wanted this to work reliably, we'd need to figure out how to install this feature conditioned on the host OS. As it currently stands, I can't see anyone trying to use this outside of an AP which should pretty much always be a compatible version of Linux, so I'm in favor of punting on this step for the time being.

Reply:
I confirmed I could not install this on my Windows machine so we should leave the installation outside of the spras environment.
```
$ pip install snakemake-executor-plugin-htcondor/
Processing c:\users\agitter\desktop\madison\collaborators\ritz\snakemake-executor-plugin-htcondor
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
ERROR: Could not find a version that satisfies the requirement htcondor<24.0.0,>=23.4.0 (from snakemake-executor-plugin-htcondor) (from versions: none)
ERROR: No matching distribution found for htcondor<24.0.0,>=23.4.0
```